src/GameFace/Utils/UsersExtractions.cs

```diff
@@ -0,0 +1,91 @@
+﻿using GameFace.Controllers;
+using GameFace.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GameFace.Utils
+{
+    public static class UsersExtractions
+    {
+
+        public static int GetExperienceUserByTask(int user, int task)
+        {
+            using (var database = new GameFaceContext())
+            {
+                int experience = 0;
+                var users = database.Users.Include(x => x.xP).Where(x => x.id == user);
+                foreach (var u in users)
+                {
+                    experience += (from xp in u.xP where xp.idTask == task select database.Tasks.Single(t => t.id == xp.idTask) into tasks select tasks.value).Sum();
+                }
+                return experience;
+            }
+        }
+
+        public static UsersAchievements CheckAvailableAcievementsForXP(int user, int task)
```

Comment:
The intention of this method isn't crystal clear to me. It seems that we are doing 2 things - calculating the new xp and returning a new achievement if something was unlocked. Generally, we want methods/classes to do only 1 thing (single responsibility principle). 

I recommend that we change the UserExtractions class to have the following methods
GetCurrentXp(int userId, int taskId) - Gets the total xp for a specific task / specific user.
GetTaskValue(int taskId) - Gets the value for the specified task.
GetNextAchievement(int userId, int taskId) - Gets the next achievement that the user would unlock

With these 3 methods we can then (at the point of posting xp) calculate the new xp and decide if the user should unlock the achievement.

The benefits of having small, single purpose methods are
1) Reuse - since we do only one thing, we could use this method in several places where we need similar functionality.
2) Intent - the intent of single purpose methods are usually more clear than methods that do several things.
3) Maintenance - single purpose methods are generally simpler and therefore easier to maintain.